include/cuco/detail/utility/cuda.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2021-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+
+#pragma once
+
+#include <cuco/detail/utility/math.hpp>
+
+namespace cuco {
+namespace detail {
+
+using index_type = int64_t;  ///< CUDA thread index type
+
+static constexpr int32_t CUCO_DEFAULT_BLOCK_SIZE = 128;  ///< Default block size
+static constexpr int32_t CUCO_DEFAULT_STRIDE     = 1;    ///< Default stride
```

Comment:
```suggestion
constexpr int32_t default_block_size() noexcept { return 128; }  ///< Default block size
constexpr int32_t default_stride() noexcept { return 1; }        ///< Default stride
```

The all-caps + `CUCO_` prefix naming scheme only applies to macros.
I tend to use a `constexpr` function over a variable here.
If we go with the variable instead, we should make them `inline constexpr` so all TUs see the same definition.
Not that for the function approach, `inline` is already implied.